aea/helpers/dialogue/base.py

```diff
@@ -178,16 +178,26 @@ def outgoing_extend(self, message: "Message") -> None:
         :param message: a message to be added
         :return: None
         """
-        self._outgoing_messages.extend([message])
+        if self.is_valid_next_message(message):
```

Comment:
what if it is not valid? presumably we should raise a `ValueError`

Reply:
Otherwise the failure mode is silent...

Reply:
The problem is, if it can fail, we need to change the public api..so I would revert and instead add a second public api which does the check and raises on error and is called `try_outgoing_extend`

Reply:
I like the idea of creating a second pair of extend. 
The new ones could extend the dialogue with the new message if the message is valid. Returns true if it does, false if it doesn't.

Reply:
Done